src/SansDaemonServiceProvider.php

```diff
@@ -55,8 +56,14 @@ protected function configureQueue()
      */
     protected function registerWorkCommand()
     {
-        $this->app->extend('command.queue.work', function ($command, Application $app) {
-            return new WorkCommand($app['queue.sansDaemonWorker'], $app['cache.store']);
-        });
+        if (version_compare($this->app->version(), "9.0.0", ">=")) {
```

Comment:
Thanks for bringing this up @arxeiss, I totally missed this.

This is great. I would like to propose an alternate solution. Instead of hardcoding a specific Laravel version and using version compare, why don't we just use a list of the known base command bindings and only register the custom `WorkCommand` if the base command has been bound to the container? I'll show you what I mean:

![carbon](https://user-images.githubusercontent.com/11756231/157240142-e33cf5b9-9698-40a6-8416-c27fb0374026.png)

This way, we won't have to hardcode a specific version and we can add more commands to the list if/when a new one is added to laravel.

Reply:
I understand what you mean. However:

1. Understanding this code is much harder than 1 IF with version compare.
2. If you would like to add more commands, you still need to copy the whole `foreach`. Because `WorkCommand` is only for `php artisan queue:work`. Other commands related with `queue` have separate class. And different classes might use different arguments. So I cannot imagine how this could be reused.

Reply:
We're not registering all queue commands, just the WorkCommand. The list is not a list of all the queue commands but a list of bindings for the WorkCommand that Laravel has used so far. If for any reason Laravel adds a new way to bind the WorkCommand to the container in a later version, we'll just add that binding to the list instead of adding an additional IF with another version compare.

With version_compare, the complexity increases with time depending on how quickly the bindings for WorkCommand changes. If the command changes in Laravel 10 for instance, we'll need to add a new IF condition and version compare just to register that binding.

Reply:
Oh right.. now I understand what you mean. Was confused a bit.
I'm OK with that, feel free to update my PR.

Just 1 little fix, I would change the order in array from newest to oldest.
Imagine, that new Laravel code/another library register something with `command.queue.work` and you would override it with your code.

So I would rather see always checking the newest, and then go to the older ones. And when you already bound something, `break` from `foreach` loop. Will prevent some unintentional bugs and safe some computer time. When you expect, developers uses the newest Laravel version.